src/app/modules/form/admin-form/admin-form.payments.service.ts

```diff
@@ -0,0 +1,83 @@
+import mongoose from 'mongoose'
+import { errAsync, okAsync, ResultAsync } from 'neverthrow'
+
+import { PaymentsUpdateDto, PaymentType } from '../../../../../shared/types'
+import { IEncryptedFormDocument } from '../../../../types'
+import { paymentConfig } from '../../../config/features/payment.config'
+import { createLoggerWithLabel } from '../../../config/logger'
+import { getEncryptedFormModel } from '../../../models/form.server.model'
+import { transformMongoError } from '../../../utils/handle-mongo-error'
+import { PossibleDatabaseError } from '../../core/core.errors'
+import { InvalidPaymentAmountError } from '../../payments/payments.errors'
+import { FormNotFoundError } from '../form.errors'
+
+const logger = createLoggerWithLabel(module)
+const EncryptedFormModel = getEncryptedFormModel(mongoose)
+
+/**
+ * Update the payments field of the given form
+ * @param formId the id of the form to update the end page for
+ * @param newPayments the new payments field to replace the current one
+ * @returns ok(updated payments object) when update is successful
+ * @returns err(FormNotFoundError) if form cannot be found
+ * @returns err(PossibleDatabaseError) if start page update fails
+ * @returns err(InvalidPaymentAmountError) if payment amount exceeds MAX_PAYMENT_AMOUNT
+ */
+export const updatePayments = (
+  formId: string,
+  newPayments: PaymentsUpdateDto,
+): ResultAsync<
+  IEncryptedFormDocument['payments_field'],
+  PossibleDatabaseError | FormNotFoundError | InvalidPaymentAmountError
+> => {
+  const { enabled } = newPayments
+
+  // Check if payment amount exceeds maxPaymentAmountCents or below minPaymentAmountCents if the payment is enabled
+  if (enabled && newPayments.payment_type === PaymentType.Fixed) {
```

Comment:
> it may be better to put validation as a separate function

Agreed. With `ByProducts` this will get way too chunky.

> Alternatively, is it possible to use joi for these validations? for e.g. (need to be tested)

I think `joi` is used for `routes` -joi-middleware-> `controller`. While services can be called by other services/controller.

Reply:
Sorry for the confusion, what I meant is to move the validation to the route validation portion, instead of adding another layer of validation in this function.

Reply:
It is already validated at the routes [here](https://github.com/opengovsg/FormSG/pull/6463/files#diff-f46e4b758b65d2a6eed5f926a370dcbdbd3bf032308ffae0ecd0e45600f529d2R315).

This is a service-level check where the intention is to further check on config-level validators (`paymentConfig.minPaymentAmountCents`) which is out of `joi` and `db` role

Reply:
i was wondering if we could add all checks, in the joi validation, as it may cause confusion as to why there're various places for input validation checks. Also, at the very minimum, i think the min/max amount check in line 48 can be moved to joi validation?

Reply:
Added here 6bf2024b6. This suggestion made the error response look really good 👍 

**Before**
`{"message":"Invalid payment amount"}`

**After**
`{"statusCode":400,"error":"Bad Request","message":"Validation failed","validation":{"body":{"source":"body","keys":["amount_cents"],"message":"\"amount_cents\" must be less than or equal to 100000000"}}}`

Reply:
should we remove this part then? since it's already in joi validation now

Reply:
The service should still check, as it can be called by other services/controller